lib/limits.c

```diff
@@ -60,18 +62,15 @@ static int setrlimit_value (unsigned int resource,
 		 * Also, we are limited to base 10 here (hex numbers will not
 		 * work with the limit string parser as is anyway)
 		 */
-		char *endptr;
-		long longlimit = strtol (value, &endptr, 10);
-		if ((0 == longlimit) && (value == endptr)) {
+		l = strtol(value, &endptr, 10);
```

Comment:
I know you are only converting the old code, but since you are touching it I think you should check `errno` before doing anything else.

Reply:
Done.